py/torch_tensorrt/dynamo/conversion/converter_utils.py

```diff
@@ -250,22 +252,113 @@ def get_trt_tensor(
     Returns:
         A TensorRT ITensor that represents the given value.
     """
-    # TRT can not add constant for bool type. We do a work around to 1) cast it to int and 2)cast to bool later
-    # This is useful for logical operations which require input to be bool type
-    if isinstance(input_val, bool):
-        input_val = int(input_val)
-    elif isinstance(input_val, torch.Tensor) and (
-        input_val.dtype == torch.bool or input_val.dtype == torch.int64
-    ):
-        input_val = input_val.to(torch.int32)
-    elif isinstance(input_val, np.ndarray) and (
-        input_val.dtype == np.bool_ or input_val.dtype == np.int64
-    ):
-        input_val = input_val.astype(np.int32)
-
-    if isinstance(input_val, (torch.Tensor, np.ndarray, int, float)):
+    # If the input is 64-bit, cast it to 32-bit for TRT freezing
+    if isinstance(input_val, torch.Tensor):
+        if input_val.dtype == torch.int64:
+            input_val = input_val.to(torch.int32)
+        elif input_val.dtype == torch.float64:
+            input_val = input_val.to(torch.float32)
```

Comment:
Augment `TRTNetwork` to store user settings

- Conversion Context class --> TRTNetwork, CompileSettings
- INetwork + Context Class, Network inputs // CTX inputs
- `ctx.net` replace network calls in converters